src/shared/constants.h

```diff
@@ -79,6 +79,14 @@ static const double BALL_MAX_RADIUS_METERS = 0.0215;
 // cover more than 20% of the ball
 static const double MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT = 0.2;
 
+// The radius of a circle region where ball placement is acceptable (in meters).
+static const double BALL_PLACEMENT_TOLERANCE_RADIUS_METERS = 0.15;
+// The radius of the outer region where robots are not allowed to be during ball
+// placement (in meters)
+static const double BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS = 0.5;
+// The time limit for ball placement in seconds
+static const int BALL_PLACEMENT_TIME_LIMIT_S = 30;
```

Comment:
Interestingly enough, it seems that our `constants.h` file is wildly inconsistent with its use of `static const` vs `constexpr`. From my knowledge, there is no real performance difference here aside from `constexpr` potentially preventing others in the future from assigning a non-compile time constant value to these variables. 

TLDR: ask the others about as I am not sure why this is the case.

Reply:
As the info I have learned just then, `constexpr` might be used in this case, but I will confirm with others.

`static const` seems to be used more in class-level constants. If we use `constexpr`, it allows complie-time optimization.

If confirmed and refactor needed, do you mind if I do this refactor in a separate ticket, so we don't mess up other features.

Reply:
Sure, sounds good.

IMO `constexpr` and `static const` in this case should not technically have a difference in optimization in this case as most of the `static const` values we have can be determined at compile-time which results in the optimization process being carried through anyways (to my understanding). This would mean the main advantage of switching would purely be to deter others in the future from using non-compile time derivable values (as `constexpr` would throw an error). You may want to refer to the link below for a clearer explanation:

https://stackoverflow.com/questions/41125651/constexpr-vs-static-const-which-one-to-prefer

Reply:
Yes, we could open a new issue to make all these constants `constexpr`. The main benefit would be so that we can use these constants in other `constexpr` variables and use them in template args, array declarations, etc.

Reply:
Done, fixed all 3 constants included in this PR with `constexpr`. Will send another PR on refactoring others after this is merged.